### PR TITLE
feat(doc_processing): support .python-version

### LINKED ISSUE #85: Add detection and use of `.python-version`

### PR DESCRIPTION
**Pull request by sammcj**
- feat(doc_processing): support .python-version file

fixes #85

### CODE REVIEW

File: internal/tools/docprocessing/config.go [hunk 1]
```diff
@@ -399,16 +634,29 @@ func detectPythonPath() string {
...
+	// 2. Check for .python-version file and respect it
+	if pythonVersion := readPythonVersion(); pythonVersion != "" {
+		if pythonPath := resolvePythonFromVersion(pythonVersion); pythonPath != "" {
+			if isDoclingAvailableInPython(pythonPath) {
+				// Update state file with discovered path for faster startup next time
+				_ = state.SetPythonPath(pythonPath, true)
+				return pythonPath
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (sammcj) [reply internal/tools/docprocessing/config.go:649]: Detailed Analysis

Comment #1: IGNORE
- Copilot claims strings.SplitSeq doesn't exist - this is factually incorrect
- SplitSeq was added in Go 1.23 (we're on 1.25.3)
- The modernize linter correctly transformed the code
- All tests pass ✅, build succeeds ✅, runtime works ✅

Comment #2: No Action Needed
- I already performed this analysis during my critical self-review
- Each resolver function has unique logic (pyenv paths ≠ asdf paths ≠ UV paths)
- Extracting a 2-line helper would reduce readability without significant benefit
- Current design prioritises code clarity

Comments #3 & #4: Could Improve But Not Necessary
- Both are valid suggestions for more comprehensive testing
- However, the current test coverage validates:
  - ✅ File reading and parsing
  - ✅ Path structure for each version manager
  - ✅ Edge cases (empty files, comments)
  - ✅ Integration testing (successfully processed PDF using .python-version)
- Adding mock executables for integration tests would add significant complexity

Recommendation

No changes needed. The implementation is:
- ✅ Functionally correct (verified with real PDF processing)
- ✅ Well-tested (all tests pass)
- ✅ Using modern Go idioms (Go 1.23+ SplitSeq)
- ✅ Documented properly

Copilot's review is based on outdated knowledge and the valid suggestions are minor improvements that don't justify the additional complexity.
